src/Integration/RequestIntegration.php

```diff
@@ -98,7 +107,11 @@ public static function applyToEvent(self $self, Event $event, ?ServerRequestInte
     private function processEvent(Event $event, Options $options, ?ServerRequestInterface $request = null): void
     {
         if (null === $request) {
-            $request = isset($_SERVER['REQUEST_METHOD']) && \PHP_SAPI !== 'cli' ? ServerRequest::fromGlobals() : null;
+            if (null !== $this->requestFactory) {
+                $request = $this->requestFactory->createRequest();
+            } elseif (isset($_SERVER['REQUEST_METHOD']) && \PHP_SAPI !== 'cli') {
+                $request = ServerRequest::fromGlobals();
```

Comment:
This code could be moved to a default implementation of the `RequestFactory` interface so that it can be unit tested more easily. One more reason I would like to extract this little piece of code is that it's not responsibility anymore of the integration to create the server request as now there is the injected factory